### PR TITLE
This makes structure tests build in bazel.

### PR DESCRIPTION
**Pull request by dlorenc**
One manual change, to workaround the gorilla/mux issue.

### CODE REVIEW

File: vendor/github.com/gorilla/mux/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# gazelle:ignore
```
Comment (nkubala) [inline vendor/github.com/gorilla/mux/BUILD.bazel]: just curious, why doesn't this work?

Comment (dlorenc) [reply vendor/github.com/gorilla/mux/BUILD.bazel]: There's a bug in the way gazelle handles go build tags. That package has two implementations of a feature inside it, one is used with go versions < 1.7, the other with >= 1.7. Gazelle doesn't understand that and breaks :(

Comment (nkubala) [reply vendor/github.com/gorilla/mux/BUILD.bazel]: ahh, makes sense.

**Comment by dlorenc** [comment]
Just rebased. PTAL

**Pull request merged**